NuRadioReco/modules/channelReadoutWindowCutter.py

```diff
@@ -0,0 +1,171 @@
+                    "Channels has only {} samples but {} samples are requested.").format(
+                    trace.shape[0], number_of_samples))
+                raise AttributeError
+
+            # windowed_channel = get_empty_channel(
+            #     station.get_id(), channel.get_id(), detector, trigger, sampling_rate)
+            # windowed_channel.add_to_trace(channel)
+
+            # channel.set_trace(windowed_channel.get_trace(), windowed_channel.get_sampling_rate())
+            # channel.set_trace_start_time(windowed_channel.get_trace_start_time())
+
+            channel_id = channel.get_id()
+            pre_trigger_time = trigger.get_pre_trigger_time_channel(channel_id)
+
+            pre_trigger_time_channel = trigger_time - pre_trigger_time - channel.get_trace_start_time()
+            
+            # throw warinings or errors depending on if the readout window is outside or too far outside the trace
+            trace_length = len(trace)
+            if pre_trigger_time_channel < -pre_trigger_time:
+                msg = ("Start of the readout window is more than pre_trigger_time before the start of the "
+                       "trace (trigger time = {}ns, pre-trigger time = {}ns, start of trace {}ns, requested "
+                       "time before trace = {}ns), this would result in rolling over the edge of the trace "
+                       "and is not the intended use of this function").format(
+                            trigger_time, pre_trigger_time, channel.get_trace_start_time(), pre_trigger_time_channel)
+                logger.error(msg)
+                raise AttributeError(msg)
+            elif pre_trigger_time_channel < 0:
+                msg = ("Start of the readout window is before the start of the trace (trigger time = {}ns, "
+                       "pre-trigger time = {}ns, start of trace {}ns, requested time before trace = {}ns), "
+                       "the trace will be rolled over the edge to fit in the readout window").format(
+                            trigger_time, pre_trigger_time, channel.get_trace_start_time(), pre_trigger_time_channel)
+                logger.warning(msg)
+            elif trigger_time > channel.get_trace_start_time() + trace_length / sampling_rate:
+                msg = ("Trigger time is after the end of the trace (trigger time = {}ns, end of trace {}ns), "
+                       "this would result in rolling over the edge of the trace and is not the intended use "
+                       "of this function").format(
+                            trigger_time, channel.get_trace_start_time() + trace_length / sampling_rate,
+                            pre_trigger_time_channel + number_of_samples / sampling_rate - trace_length / sampling_rate)
+                logger.error(msg)
+                raise AttributeError(msg)
+            elif pre_trigger_time_channel + number_of_samples / sampling_rate > trace_length / sampling_rate:
```

Comment:
I think you have to use  `detector_sampling_rate` here `number_of_samples / sampling_rate`